Fait rougir travis en cas d'erreur d'un test

A-312 opened the pull request:
Les tests ne retourne pas d'erreur en cas d'échec

# Q/A : 

Il faut merge cette PR, et rebase les 2 PR suivante dans l'état et voir si travis rougi (rate)
 - https://travis-ci.org/zestedesavoir/zds-site/jobs/535731371#L2259
 - https://travis-ci.org/zestedesavoir/zds-site/jobs/537493898#L2497

Travis réussi alors qu'il ne devrait pas, s'il rate c'est bon signe.

A-312 committed:
Ajout d'une fonction qui fait rougir travis en cas d'erreur

A-312 committed:
Ajoute zds_fold_end pour simplifier l'affichage

L'erreur détaillée s'affichera dans le folder (il faudra dérouler l'élément)

A-312 committed:
Ajout des gateway pour faire rougir Travis

Certain test ne bloquait pas le job en cas d'échec. Cette PR permet d'intercepter les jobs en questions

A-312 commented:
Une erreur devrait être corrigé par : #5349 mais je ne comprends pas la deuxième dans https://travis-ci.org/zestedesavoir/zds-site/jobs/537951812#L2260

artragis commented:
ça dit juste qu'il faut enlever le "e" en argument ici : https://github.com/zestedesavoir/zds-site/blob/dev/assets/js/editor-persistence.js#L73

artragis commented:
tu peux rebase stp?

artragis commented:
tu pourras fixer les trois petits soucis de pep8 stp?

A-312 commented:
pep8 ? est-ce lint backend ?

```
* Run lint for backend
./zds/tutorialv2/publication_utils.py:9:1: F401 'os.mkdir' imported but unused
./zds/tutorialv2/api/views.py:80:1: W293 blank line contains whitespace
./zds/tutorialv2/management/commands/publication_watchdog.py:14:1: E302 expected 2 blank lines, found 1
```

artragis commented:
oui

A-312 committed:
Fix lint
'os.mkdir' imported but unused
blank line contains whitespace
expected 2 blank lines, found 1
change outer quotes to avoid escaping inner quotes
line break after binary operator

artragis commented:
faudrait que tu rebase stp.

artragis commented on scripts/travis_script.sh:
```diff
@@ -28,22 +32,28 @@ if [[ "$1" == "lint_backend" ]] && [[ "$ZDS_TEST_JOB" == *"zds.gallery"* ]]; the
         ./scripts/no_import_zds_settings.sh \
             && flake8 \
-            && flake8 --config=zds/settings/.flake8 zds/settings
+            && flake8 --config=zds/settings/.flake8 zds/settings; exVal=$?
```
`--config=zds/settings/.flake8` pas bon, faut mettre .flake8 tout court.

A-312 replied:
je l'avais changé volontairement il me semble

artragis replied:
je vois pour les settings on a un flake8 propre. Ok faut donc ajouter les bonnes constantes dans le .flake8

A-312 replied:
Je mets tout pareil ?

A-312 replied:
Autant remettre le même

A-312 replied:
> je l'avais changé volontairement il me semble

En y repensant, je ne sais pas si j'y ai touché

artragis commented:
pourquoi tu as supprimé le fichier?

A-312 committed:
Même ignore que zds-site/.flake8

A-312 committed:
Utilise gateway pour exit en cas d'erreur

A-312 committed:
Fix lint W503 line break before binary operator
